doc/sourmash-internals.md

```diff
@@ -0,0 +1,648 @@
+sketches are used as a picklist to retrieve the desired signatures.
+
+Thus, while standalone manifests can point at any kind of container,
+including JSON files or LCA databases, they are most efficient when
+`internal_location` points at a file with either a single sketch in
+it, or a manifest that supports direct loading of sketches. Therefore,
+we suggest using standalone manifest indices.
+
+Note that searching a standalone manifest is currently done through a
+linear iteration, and does not use any features of indexed containers
+such as SBTs or LCAs.  This is fine for `gather` with the default
+approach, but is probably suboptimal for a `search`.
+
+### Pathlists and `--from-file`
+
+All (or most) sourmash commands natively support taking in lists of
+signature collections via pathlists, `--from-file`, or paths to
+directories. This is useful for situations where you have thousands of
+signature files and don't want to provide them explicitly on the
+command line; you can simply put a list of the files in a text file,
+and pass it in directly (or use `--from-file` to pass it in).
+
+Both pathlists and files passed to `--from-file` contain a list of
+paths to be loaded; relatives paths will be interpreted relative to
+the current working directory of sourmash.  Pathlists should be
+universally available on sourmash commands.  When `--from-file` is
+available for a command, sourmash will behave as if the file paths in
+the file were provided on the command line.
+
+We suggest avoiding pathlists. Instead, we suggest using `--from-file`
+or a standalone manifest index (generated with `sourmash sig
+collect`). This is because the signatures from pathlists are loaded
+into memory (see `MultiIndex`, above) it is generally a bad idea to
+use them - they may be slow to load and may consume a lot of
+memory. They also do not support good loading error messages;
+see [sourmash#1414](https://github.com/sourmash-bio/sourmash/issues/1414).
+
+### Extensions for outputting index classes
+
+Most commands that support saving signatures will save them in a
+variety of formats, based on the extension provided (see
+[sourmash#1890](https://github.com/sourmash-bio/sourmash/issues/1890)
+for exceptions). The supported extensions are -
+
+* `.zip` for `ZipFileLinearIndex`
+* `.sqldb` for `SqliteIndex`
+* `.sig` or `.sig.gz` for JSON/gzipped JSON
+* `dirname/` to save in a directory hierarchy
+
+The default signature save format is JSON, if the extension is not
+recognized.
+
+## Speeding up `gather` and `search`
+
+There are two primary search commands in sourmash: `gather` and
+`search`.
+
+`gather` calculates a minimum metagenome cover as discussed in [Irber et al., 2022](https://www.biorxiv.org/content/10.1101/2022.01.11.475838v2). It
+is mostly intended for querying a database with a metagenome, although
+it can be used with genome queries, as well. It depends on overlap
+analyses and can only be used with FracMinHash sketches.
```

Comment:
Is the overlap analysis containment or overlap coefficient(i.e. max containment)?

Reply:
neither, actually - number of shared bases (which can be used to calculate both containment index and max containment).

Reply:
committing:
>This approach relies on overlaps between genomes and metagenomes and can only be used with FracMinHash sketches.

Reply:
updated in fbbcee56